tests/ui/lint/non_local_definitions.rs

```diff
@@ -461,6 +461,17 @@ non_local_macro::non_local_impl!(CargoUpdate);
 non_local_macro::non_local_macro_rules!(my_macro);
 //~^ WARN non-local `macro_rules!` definition
 
+// https://github.com/rust-lang/rust/issues/123573#issue-2229428739
+pub trait StarlarkTypeRepr {}
+
+impl<'a, T: 'a> StarlarkTypeRepr for &[T] where &'a T: StarlarkTypeRepr {}
```

Comment:
Why is this in like one mega UI test? I feel like it should he broken up into several files.

Also like, there's no reason for us to put crate-specific names on these traits. I think a placeholder name is better than StarlarkXYZ for a UI test.

Reply:
> Why is this in like one mega UI test? I feel like it should he broken up into several files.

Because that's currently the only UI test file for the `non_local_definitions` lint. But you're right this file is now too big, I'll split in a follow-up PR.

> Also like, there's no reason for us to put crate-specific names on these traits. I think a placeholder name is better than StarlarkXYZ for a UI test.

My reasoning is recognition (of the crate) and ease of navigation (in this file).

Reply:
please split this file in this PR. I would also prefer multiple self-contained tests

Reply:
It took me took a bit of time but I ended up splitting it into 11 different UI test files.
It's not perfect but it's definitively much better. (I also reverted to the cratename-less repro).